app/controllers/application_drafts_controller.rb

```diff
@@ -111,14 +122,20 @@ def ensure_max_applications
     end
   end
 
+  def require_student
+    unless application_draft.as_student?
+      redirect_to application_drafts_path, alert: 'You must be listed as a student on your team'
+    end
+  end
+
   def current_team
-    current_student.current_team || coaches_team
+    current_student.current_team || extended_team
   end
 
-  def coaches_team
-    @coaches_team ||= begin
+  def extended_team
```

Comment:
I am so lost with this method, I have no idea what's going on, could we consider a refactor?